### PR TITLE
Merge #127 into master along with fixes

### PR DESCRIPTION
**Pull request by SolidHal**
this brings in a saner method to detect encryption from @tslilc and fixes from @rk-zero and myself

### CODE REVIEW

**Comment by SolidHal** [comment]
Will merge once I have a chance to test

File: scripts/InstallScripts/UpgradeKernel.sh [hunk 1]
```diff
@@ -30,6 +30,7 @@ echo
 if [[ $REPLY =~ ^[Yy]$ ]]
 then
     echo Writing kernel partition
+    dd if=/dev/zero of=/dev/mmcblk2p1 bs=1 count=65536
```
Comment (rk-zero) [inline scripts/InstallScripts/UpgradeKernel.sh]: Is this really correct? I replaced the 32MB empty kernel file with a line like that, but the bs should be 512, shouldn't it?

I'm so proud you actually integrated the changes!!

Comment (SolidHal) [reply scripts/InstallScripts/UpgradeKernel.sh]: You're right, bs definitely should be 512 bytes

Comment (SolidHal) [reply scripts/InstallScripts/UpgradeKernel.sh]: thanks for all of your work @rk-zero!

**Pull request merged**